### PR TITLE
LPS-170904 Use the admin set configuration for explicitConsentMode in CookiesManagerImpl

### PR DESCRIPTION
**Pull request by oliverkecskemety**
[LPS-170904](https://issues.liferay.com/browse/LPS-170904)

### CODE REVIEW

**Comment by oliverkecskemety** [comment]
ci:test:sf

**Comment by oliverkecskemety** [comment]
ci:test:relevant

**Comment by JoyceWang08211** [comment]
ci:reevaluate:1327401_7507

**Comment by JoyceWang08211** [comment]
ci:test:relevant

**Comment by JoyceWang08211** [comment]
ci:test:bundle

**Comment by JoyceWang08211** [comment]
Reported the bug in https://issues.liferay.com/browse/LPS-171570.
cc @oliverkecskemety

**Comment by gaborlovasdotcom** [comment]
ci:test:cookies

File: modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/manager/CookiesManagerImpl.java [hunk 1]
```diff
@@ -287,7 +290,41 @@ else if (consentType == CookiesConstants.CONSENT_TYPE_PERSONALIZATION) {
...
+						cookiesPreferenceHandlingConfiguration =
+							_configurationProvider.getGroupConfiguration(
+								CookiesPreferenceHandlingConfiguration.class,
+								groupId);
+					}
+					else {
+						long companyId = _portal.getCompanyId(
+							httpServletRequest);
+
+						if (companyId > 0) {
```
Comment (stian-sigvartsen) [inline modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/manager/CookiesManagerImpl.java:312]: Did you find that `companyId` ever 0?

Comment (KOliver94) [reply modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/manager/CookiesManagerImpl.java:312]: Good point. It's unnecessary, this part should catch it before: https://github.com/liferay-appsec/liferay-portal/blob/10ff0c6bd7dcd01f0d1f563d3e652530d0239d37/portal-impl/src/com/liferay/portal/util/PortalInstances.java#L146-L152

**Comment by stian-sigvartsen** [comment]
Looks good to me. Just made some reduction to the indentation to improve readability and avoiding unnecessary configuration retrieval.

**Comment by oliverkecskemety** [comment]
ci:stop

**Comment by oliverkecskemety** [comment]
ci:test:sf

**Comment by oliverkecskemety** [comment]
ci:test:relevant

**Comment by oliverkecskemety** [comment]
ci:test:cookies

File: modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/manager/CookiesManagerImpl.java (hunk 1 above)
Comment (oliverkecskemety) [reply modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/manager/CookiesManagerImpl.java]: Good point. It's unnecessary, this part should catch it before: https://github.com/liferay-appsec/liferay-portal/blob/10ff0c6bd7dcd01f0d1f563d3e652530d0239d37/portal-impl/src/com/liferay/portal/util/PortalInstances.java#L146-L152

**Comment by gaborlovasdotcom** [comment]
ci:test:bundle

**Comment by oliverkecskemety** [comment]
ci:test:sf

**Comment by oliverkecskemety** [comment]
ci:test:cookies

**Comment by oliverkecskemety** [comment]
ci:test:sf

**Comment by oliverkecskemety** [comment]
ci:test:cookies

**Comment by oliverkecskemety** [comment]
ci:test:cookies

**Comment by oliverkecskemety** [comment]
ci:test:sf

**Comment by oliverkecskemety** [comment]
ci:stop

**Comment by oliverkecskemety** [comment]
ci:test:sf

**Comment by oliverkecskemety** [comment]
ci:test:cookies

**Comment by oliverkecskemety** [comment]
ci:test:relevant

**Comment by oliverkecskemety** [comment]
ci:test:cookies

**Comment by JoyceWang08211** [comment]
The failure of [LocalFile.CookiesConsentPanel#CookiesConsentPanelCouldBeEnabledAndCustomized](https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/161346//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_CookiesConsentPanel_CookiesConsentPanelCouldBeEnabledAndCustomized_) is not related to the current changes. It also appears on the latest upstream [run](https://testray.liferay.com/home/-/testray/case_results/2190827310). And it was reported in https://issues.liferay.com/browse/LPS-173948.
cc @oliverkecskemety

**Comment by stian-sigvartsen** [comment]
Hi @oliverkecskemety . This is looking really good!
The only thing I am wondering about is why bb8846844 is better? And if that commit is necessary?
It follows on from my earlier concern about an untracked `Map` being returned from `_getCookiesMap()` when `HttpServletRequest` is null. Not tracked in the `HttpServletRequest` (obviously :) ) which the caller might be assuming it will be.

**Comment by oliverkecskemety** [comment]
Hi @stian-sigvartsen, thanks for the review!

I understand you concerns however I think this would be the easiest and most maintainable solution. My solutions is based on the following parts:
https://github.com/liferay-appsec/liferay-portal/blob/ae403c678eca01bfcf819db7b9b156d1f1cdd957/modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/manager/CookiesManagerImpl.java#L454-L466
The problem is that Lines 454-456 will throw an NPE as `httpServletRequest` is `null`. However, if we go further and ignore this problem we would try to get the cookies from the `httpServletRequest` which would also fail with an NPE but considering we have no request that means no cookies and in this case we would return a `new HashMap<>()`.

`_getCookiesMap()` is used in 3 places one of them would not be called when `httpServletRequest == null`. The other one is in `deleteCookies()` function when we try to delete from this map which I don't think is a problem. And the last one is in `_getCookieValue()` where we try to find a cookie inside the map and return its value or `null` if not found.

If we don't change `_getCookiesMap()` we need to handle each case separately with checks. You can see I removed two of them because I would need to add one more to `_deleteCookieConsentCookies()` or `getCookieValue()` which means nearly all public functions have a check on `httpServletRequest == null` even if all issues are caused by `_getCookiesMap()`.

If you think it's not the right way to fix the issue I can change is obviously. :)

**Comment by stian-sigvartsen** [comment]
Thanks @oliverkecskemety for taking the time to explain your reasoning! Such usability arguments are not always very obvious, but it makes perfect sense. I've pushed a commit which makes the map immutable, which sounds like will be fine given the 3 cases?

**Comment by oliverkecskemety** [comment]
Thank you! It should cover everything.

**Comment by stian-sigvartsen** [comment]
ci:test:sf

**Review by stian-sigvartsen — APPROVED** [review]

**Comment by melodywu** [comment]
ci:test:bundle

**Review by gaborlovasdotcom — APPROVED** [review]

**Comment by oliverkecskemety** [comment]
ci:forward

**Comment by oliverkecskemety** [comment]
ci:forward

**Pull request closed without merging**